### PR TITLE
Restructure storage directory

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/339

### CODE REVIEW

**Comment by JinmingHu-MSFT** [comment]
@danieljurek @Azure/azure-sdk-eng Hi, can you take a look at the eng/sys related failure? We restructured the directory hierarchy and this error comes up.

**Comment by JinmingHu-MSFT** [comment]
@danieljurek @Azure/azure-sdk-eng Hi, I made some changes to eng/sys code, can you take time to review it? Thanks

File: sdk/storage/CMakeLists.txt [hunk 1]
```diff
@@ -8,174 +8,31 @@ set(CMAKE_CXX_STANDARD 14)
...
-
-add_library(azure-storage-file-share ${AZURE_STORAGE_SHARES_HEADER} ${AZURE_STORAGE_SHARES_SOURCE})
-target_link_libraries(azure-storage-file-share azure-storage-common)
-
-add_library(azure::storage::common ALIAS azure-storage-common)
-add_library(azure::storage::blob ALIAS azure-storage-blob)
-add_library(azure::storage::file::datalake ALIAS azure-storage-file-datalake)
-add_library(azure::storage::file::share ALIAS azure-storage-file-share)
 
+option(BUILD_STORAGE_SAMPLES "Build storage sample codes" ON)
```
Comment (RickWinter) [inline sdk/storage/CMakeLists.txt:26]: Nit, place the options at the top of the CMakeList so they are easy to find

File: sdk/storage/azure-storage-blobs/version.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+1.0.0-beta.1
```
Comment (RickWinter) [inline sdk/storage/azure-storage-blobs/version.txt:1]: Get rid of this file and use a version.h file in the `azure/storage/blobs/version.h`.  See azure-core folder for the contents of that file.

File: sdk/storage/azure-storage-common/version.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+1.0.0-beta.1
```
Comment (RickWinter) [inline sdk/storage/azure-storage-common/version.txt:1]: See comment about using a version.h file above.  There needs to be a version.h file in the `inc/azure/<service>/` folder for each service

File: sdk/storage/azure-storage-files-datalake/version.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+1.0.0-beta.1
```
Comment (RickWinter) [inline sdk/storage/azure-storage-files-datalake/version.txt:1]: See comment on version.h files

File: sdk/storage/azure-storage-blobs/version.txt (hunk 2 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/azure-storage-blobs/version.txt:1]: https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/version.txt

Comment (JinmingHu-MSFT) [reply sdk/storage/azure-storage-blobs/version.txt:1]: azure core folder also has this file.

Comment (JinmingHu-MSFT) [reply sdk/storage/azure-storage-blobs/version.txt:1]: I think we should save the version information only at one place, either in version.txt or in version.hpp. I'm fine either way.

Comment (JinmingHu-MSFT) [reply sdk/storage/azure-storage-blobs/version.txt:1]: https://github.com/Azure/azure-sdk-for-cpp/issues/543 I created an issue to track that. This is not a big deal. I think we can figure it out in the future.

File: sdk/storage/CMakeLists.txt (hunk 1 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/CMakeLists.txt]: fixed

**Pull request merged**